pkg/dashboard/resource/stream.go

```diff
@@ -0,0 +1,168 @@
+	functions, err := sr.getPlatform().GetFunctions(ctx, getFunctionsOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed getting project functions")
+	}
+
+	// iterate over functions and look for v3iostreams
+	streams, err := sr.getStreamsFromFunctions(functions)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed getting streams from functions")
+	}
+
+	return streams, nil
+}
+
+func (sr *streamResource) getStreamsFromFunctions(functions []platform.Function) (map[string]restful.Attributes, error) {
+
+	streamsMap := map[string]restful.Attributes{}
+
+	for _, function := range functions {
+		v3ioStreamsMap := functionconfig.GetTriggersByKind(function.GetConfig().Spec.Triggers, "v3ioStream")
+		for streamName, stream := range v3ioStreamsMap {
+
+			// add stream to map, with a key in the format: "function-name@stream-name"
+			keyName := fmt.Sprintf("%s@%s", function.GetConfig().Meta.Name, streamName)
+			streamsMap[keyName] = restful.Attributes{
+				"consumerGroup": stream.Attributes["consumerGroup"],
+				"containerName": stream.Attributes["containerName"],
+				"streamPath":    stream.Attributes["streamPath"],
+			}
+		}
+	}
+
+	return streamsMap, nil
+}
+
+func (sr *streamResource) getNamespaceFromRequest(request *http.Request) string {
+	return sr.getNamespaceOrDefault(request.Header.Get("x-nuclio-project-namespace"))
+}
+
+func (sr *streamResource) getProjectByName(request *http.Request, projectName, projectNamespace string) (platform.Project, error) {
+	ctx := request.Context()
+
+	requestOrigin, sessionCookie := sr.getRequestOriginAndSessionCookie(request)
+	projects, err := sr.getPlatform().GetProjects(ctx, &platform.GetProjectsOptions{
+		Meta: platform.ProjectMeta{
+			Name:      projectName,
+			Namespace: projectNamespace,
+		},
+		AuthSession: sr.getCtxSession(request),
+		PermissionOptions: opa.PermissionOptions{
+			MemberIds:           opa.GetUserAndGroupIdsFromAuthSession(sr.getCtxSession(request)),
+			RaiseForbidden:      true,
+			OverrideHeaderValue: request.Header.Get(opa.OverrideHeader),
+		},
+		RequestOrigin: requestOrigin,
+		SessionCookie: sessionCookie,
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get projects")
+	}
+
+	if len(projects) == 0 {
+		return nil, nuclio.NewErrNotFound("Project not found")
+	}
+	return projects[0], nil
+}
+
+func (sr *streamResource) getRequestOriginAndSessionCookie(request *http.Request) (platformconfig.ProjectsLeaderKind, *http.Cookie) {
+	requestOrigin := platformconfig.ProjectsLeaderKind(request.Header.Get(iguazio.ProjectsRoleHeaderKey))
+
+	// ignore error here, and just return a nil cookie when no session was passed (relevant only on leader/follower mode)
+	sessionCookie, _ := request.Cookie("session")
+
+	return requestOrigin, sessionCookie
+}
+
+// register the resource
+var streamResourceInstance = &streamResource{
+	resource: newResource("api/streams", []restful.ResourceMethod{
```

Comment:
```suggestion
	resource: newResource("api/v3io_streams", []restful.ResourceMethod{
```
Since it's a v3io-specific resource, also suggest to rename the struct, filename, etc

Reply:
It is currently a v3io-specific resource, but this might change if we would want the get similar information on other stream (kafka for example).
Perhaps we can name it `v3io_streams` and change it if the need arises?..